corehq/util/es/interface.py

```diff
@@ -113,7 +113,8 @@ def scroll(self, scroll_id=None, body=None, params=None, **kwargs):
         return results
 
     def scan(self, index_alias, query, doc_type):
-        return scan(self.es, query=query, index=index_alias, doc_type=doc_type, search_type='scan')
+        return scan(self.es, query=query, index=index_alias, doc_type=doc_type,
+                    preserve_order=True)
```

Comment:
Does `preserve_order=True` replace `search_type='scan'`? The [change log](https://www.elastic.co/guide/en/elasticsearch/reference/2.1/breaking_21_search_changes.html) mentions sorting by `_doc`; is that equivalent?

Reply:
Hmm, I thought this was the case (using `preserve_order=True` resulted in a `sort='_doc'` query), but while fetching the requisite links to confirm my understanding, I realized a mistake.

The docs for this function are extremely misleading.  Even to this day (as of version 7.x) they [reference setting `search_type` to `scan`](https://elasticsearch-py.readthedocs.io/en/v7.14.0/helpers.html?highlight=search_type#scan), despite Elasticsearch not _having_ a `scan` search type since v2.0.  I little bit of digging turned up two, two-year-old GitHub issues regarding the matter (one [still open](https://github.com/elastic/elasticsearch-py/issues/842) and one [closed](https://github.com/elastic/elasticsearch-py/issues/835)).

I think the python library completely changed the functionality of this `preserve_order` argument from 5.x onward without changing the docs. When set to `False` (the default) in any library version <5.x, the implementation [adds the `search_type='scan'` query parameter](https://github.com/elastic/elasticsearch-py/blob/2.5.0/elasticsearch2/helpers/__init__.py#L275-L276) (even though that parameter was deprecated in ES v2.1).

In all library versions >=5.x, the implementation now [explicitly sets the query argument to `sort='_doc'`](https://github.com/elastic/elasticsearch-py/blob/5.5.5/elasticsearch5/helpers/__init__.py#L359-L361), silently clobbering any existing `query` argument specified by the caller.

I now think that python library never stopped using `search_type='scan'` in any 2.x version. It appears as though they only dropped it in v5.x, despite Elasticsearch providing an alternative way to achieve the same (performant) end result without this parameter since v2.1.  If there is no other way to achieve this without forking the `scan()` function into the HQ repo and re-writing it (not worth the effort), then I think this PR is pointless. I'll have a look.